react/features/analytics/functions.js

```diff
@@ -82,54 +77,77 @@ export function initAnalytics({ getState }: { getState: Function }) {
         version: JitsiMeetJS.version,
         whiteListedEvents
     };
+    const amplitude = new AmplitudeHandler(handlerConstructorOptions);
 
-    _loadHandlers(scriptURLs, handlerConstructorOptions)
-        .then(handlers => {
-            const roomName = state['features/base/conference'].room;
-            const permanentProperties = {};
+    return (
+        _loadHandlers(scriptURLs, handlerConstructorOptions)
+            .then(handlers => [ amplitude, ...handlers ])
+            .catch(e => {
+                analytics.dispose();
+                logger.error(e);
 
-            if (server) {
-                permanentProperties.server = server;
-            }
-            if (group) {
-                permanentProperties.group = group;
-            }
+                return [];
+            }));
+
+}
 
-            //  Report if user is using websocket
-            permanentProperties.websocket = navigator.product !== 'ReactNative' && typeof config.websocket === 'string';
-
-            // Optionally, include local deployment information based on the
-            // contents of window.config.deploymentInfo.
-            if (deploymentInfo) {
-                for (const key in deploymentInfo) {
-                    if (deploymentInfo.hasOwnProperty(key)) {
-                        permanentProperties[key] = deploymentInfo[key];
-                    }
-                }
+/**
+ * Inits JitsiMeetJS.analytics by setting permanent properties and setting the handlers from the loaded scripts.
+ * NOTE: Has to be used after JitsiMeetJS.init. Otherwise analytics will be null.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action} is being dispatched.
+ * @param {Array<Object>} handlers - The analytics handlers.
+ * @returns {void}
+ */
+export function initAnalytics({ getState }: { getState: Function }, handlers: Array<Object>) {
+    if (!analytics || !isAnalyticsEnabled(getState) || !handlers || handlers.length === 0) {
```

Comment:
How is it possible for `analytics` to be undefined? It's an import and AFAICT no other part of the code does this check.

According to the type annotation `handlers` cannot be undefined either.

Reply:
Fixed!